week-5-ci-and-production/w05e02-continuous-deployment/.github/workflows/front.yml

```diff
@@ -0,0 +1,38 @@
+name: FRONT – Stage Deploy
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'shopping-lister-front/**'
+
+jobs:
+  test_build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./shopping-lister-front
+      - name: Format code and check for errors
+        run: npm run lint:full
```

Comment:
z tego co sprawdzałem polecenie `lint` uruchomi też `prettier'a` bo config jest podpięty z tyczką do tego. Więc chyba nie ma takiej potrzeby.

Zauważ że `type-check` uruchomi sie w kroku `build`
Zauważ również że zamiast `&&` możesz skorzystać tutaj z "tricku" zrobionego właśnie w kroku `build` bo tutaj jest dointalowana paczka `npm-run-all`. `create-vue` akurat daje ją  w scaffold z automatu.

Reply:
W nowym commicie użyłem tego `tricku`, fajnie to działa – dzięki za tip ✋